api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/DiagnosticReportHandler.java

```diff
@@ -1,12 +1,19 @@
 package org.openmrs.module.fhir.api.diagnosticreport;
 
 import ca.uhn.fhir.model.dstu2.resource.DiagnosticReport;
+import org.openmrs.Obs;
+
+import java.util.List;
+import java.util.Set;
 
 public interface DiagnosticReportHandler {
-	
-	public DiagnosticReport generateFHIRDiagnosticReport(DiagnosticReportTemplate omrsDiagnosticReport);
-	
-	public DiagnosticReportTemplate generateOpenMRSDiagnosticReport(DiagnosticReport fhirDiagnosticReport);
-	
+
 	public String getId();
+
```

Comment:
@surangak I'm thinking that, when users want to implement their own handler, the only have to worry about OpenMRS APIs (OpenMRS Obs, ComplexObs etc). Services will call to Util class. Then Util class will call to appropriate Handler method.
E.g.
If a service calls to `generateOpenMRSDiagnosticReport` method in Util class, then this method will call to saveObs (which means createObs) method. It only get List of Obs.